release-to-candidate/action.yaml

```diff
@@ -43,23 +47,57 @@ runs:
         git config --global user.email "github-actions@github.com"
         git config --global user.name "Github Actions"
 
-        # Install moreutils so we have access to sponge
-        sudo apt-get update; sudo apt-get install -y moreutils
+    - name: Find the snapcraft.yaml path
+      id: yaml-path
+      shell: bash
+      run : |
+        if [[ -n "${{ inputs.snapcraft-yaml-path }}"; then
+          yaml-path="${{ inputs.snapcraft-yaml-path }}"
+        else
+          snapcraft-yaml-paths=(
+            "snap/snapcraft.yaml"
+            "snapcraft.yaml"
+            "build-aux/snap/snapcraft.yaml"
+            ".snapcraft.yaml"
+          )
+          
+          for file in "${snapcraft-yaml-paths[@]}"; do
+              if [[ -f "$file" ]]; then
+                  yaml-path="$file"
+              fi
+          done
+        fi
+        if [[ ! -n "${yaml-path}" ]]; then
+           echo "No snapcraft.yaml found" >2
+           exit 1
+        fi
+        echo "yaml-path=${yaml-path}" >> "$GITHUB_OUTPUT"
+        echo "snap-name=$(yq -r '.name' "$yaml-path")" >> "$GITHUB_OUTPUT"
+
 
     - name: Build the snap (${{ inputs.architecture }})
       id: build
       shell: bash
+      with:
+        yaml-path: ${{ steps.yaml-path.outputs.yaml-path }}
```

Comment:
This isn't correct syntax - you need to remove the `with` here, and create a `yaml_path` entry in the `env` map